consensus/types/src/runtime_var_list.rs

```diff
@@ -62,21 +82,46 @@ impl<T> RuntimeVariableList<T> {
     pub fn from_vec(mut vec: Vec<T>, max_len: usize) -> Self {
         vec.truncate(max_len);
 
-        Self { vec, max_len }
+        Self {
+            vec,
+            max_len: Some(max_len),
+        }
     }
 
-    /// Create an empty list.
+    /// Create an empty list with the given `max_len`.
     pub fn empty(max_len: usize) -> Self {
         Self {
             vec: vec![],
-            max_len,
+            max_len: Some(max_len),
         }
     }
 
     pub fn as_slice(&self) -> &[T] {
         self.vec.as_slice()
     }
 
+    pub fn as_mut_slice(&mut self) -> Option<&mut [T]> {
+        if self.max_len.is_none() {
+            return None;
+        };
+        Some(self.vec.as_mut_slice())
+    }
+
+    /// Returns an instance of `Self` with max_len = None.
+    ///
+    /// No mutating operation can be performed on an uninitialized instance
+    /// without first setting `max_len`.
+    pub fn empty_uninitialized() -> Self {
```

Comment:
feels like this function is for convenience? the little downside is having to make `max_len` optional when it's known most of the time. But I guess the convenience is worth it if we don't need to mutate it all.